### PR TITLE
EOS-17508 Fixing Decryption Changes in CSM.

### PR DESCRIPTION
**Pull request by prathameshrodi**
# Backend

## Problem Statement
<pre>
Story Ref (if any): EOS-17508
S3 password is not decrypted
</pre>
## Unit testing on RPM done
<pre>
Yes
</pre>
## Problem Description
<pre>
Currently Cortx-manager was not using Decryption on Passwords received from User.
</pre>
## Solution
<pre>
Changes Made to user decryption and Fetch Data Ip from Scripts.
</pre>
## Unit Test Cases
<pre>
Locally Tested.
</pre>
Signed-off-by:

### CODE REVIEW

**Review by avrnv — APPROVED** [review]

File: csm/conf/configure.py [hunk 1]
```diff
@@ -58,27 +58,29 @@ async def execute(self, command):
...
-                                                        'localhost'))
-                Configure._set_db_host_addr('es',
-                                            data_nw.get('private_ip',
-                                                        'localhost'))
-                Configure._set_fqdn_for_nodeid()
                 Configure._set_healthmap_path()
-                Configure._set_rmq_cluster_nodes()
+            Configure._set_node_id()
+            machine_id = Setup._get_machine_id()
+            data_nw = Configure._get_data_nw_info(machine_id)
```
Comment (pawankumarsrivastava) [inline csm/conf/configure.py:64]: Move all the static strings to const.

File: csm/core/blogic/const.py [hunk 2]
```diff
@@ -35,7 +35,7 @@
 DB_CONF_FILE_NAME = 'database.yaml'
 PLUGIN_DIR = 'cortx'
 WEB_DEFAULT_PORT = 28100 # currently being used by USL only
-PROVISIONER_LOG_FILE_PATH = "/var/log/seagate/provisioner"
+PROVISIONER_LOG_FILE_PATH = "/var/log/seagate"
```
Comment (pawankumarsrivastava) [inline csm/core/blogic/const.py:38]: Can we read this path from the KV config passed during setup?

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

File: csm/core/blogic/const.py (hunk 2 above)
Comment (prathameshrodi) [reply csm/core/blogic/const.py:38]: No this is not Parsed in Config so can't fetch from that also this path is required by CSM so access provided is in post_install command itself.

File: csm/conf/configure.py (hunk 1 above)
Comment (prathameshrodi) [reply csm/conf/configure.py:64]: sure

File: csm/conf/setup.py [hunk 3]
```diff
@@ -68,6 +69,8 @@ def _set_deployment_mode(self):
         if self._setup_info[const.NODE_TYPE] == const.VM:
             Log.info("Running Csm Setup for VM Environment Mode.")
             self._is_env_vm = True
+        if Conf.get(const.CONSUMER_INDEX, const.DEPLOYMENT_MODE) == "DEV":
```
Comment (tb-s) [inline csm/conf/setup.py:72]: Deployment mode was changed to lowercase in `templates/csm_setup_conf_template.json` but not here. I think we need to do as with the rest of the code where we use the constant defined on the all-encompassing `const` module.

Also, it is good practice to run case-insensitive checks for strings used as state enumerations.

**Review by tb-s — CHANGES_REQUESTED** [review]

File: csm/core/blogic/const.py (hunk 2 above)
Comment (prathameshrodi) [reply csm/core/blogic/const.py:38]: No this is a log path and not present in Conf Store also as per our dicussion the access for the path is required by CSM hence this should be handeled by CSM only.

File: csm/conf/configure.py (hunk 1 above)
Comment (prathameshrodi) [reply csm/conf/configure.py:64]: Made the Changes Please check.

File: csm/conf/setup.py (hunk 3 above)
Comment (prathameshrodi) [reply csm/conf/setup.py]: Done the changes. 
This was Skipped for some reason fixed by using a single key from const file.

**Review by pawankumarsrivastava — APPROVED** [review]

**Review by tb-s — APPROVED** [review]

**Review by kupranay — APPROVED** [review]

**Pull request merged**